### ISSUE TITLE
`sync`: '--root' option leads to aur-build '-r' option (aka '--rmdeps') instead of '--root'

### ISSUE THREAD

**Issue by paul-ri** [aurutils/aurutils#1148]
__Before filing a bug, check the following:__

1) Do you use the latest release of aurutils and its dependencies? If so,
is the issue reproducible from the master branch?
2) Does the package conform to 
[`PKGBUILD(5)`](https://www.archlinux.org/pacman/PKGBUILD.5.html) and the 
[AUR package guidelines](https://wiki.archlinux.org/index.php/Arch_User_Repository#Submitting_packages)?
3) Does the package provide the correct metadata on the 
[AUR RPC interface](https://aur.archlinux.org/rpc.php)?
4) Does the package build with `makepkg -s` or `extra-x86_64-build` ?
5) Does the package use internal `makepkg` functions? (see [FS#43502](https://bugs.archlinux.org/task/43502))
6) Is the problem reproducible, and not due to a misconfiguration of
`pacman`, `makepkg`, `sudoers`, `gpg` or others?

__If yes to all, create a debug log:__

```
AUR_DEBUG=1 aur <name> <arguments> >aurutils.log 2>&1
```

and attach aurutils.log to this issue.

## Description

Logs: [aurutils.log](https://github.com/aurutils/aurutils/files/14655831/aurutils.log)

I believe the following aur-sync line ([link](https://github.com/aurutils/aurutils/blob/fae585e740703786423803ef33336d8924192fd4/lib/aur-sync#L167)) should append `--root` instead of `-r` to `build_repo_args`, since this leads to aur-build's `--rmdeps` [option](https://github.com/aurutils/aurutils/blob/fae585e740703786423803ef33336d8924192fd4/lib/aur-build#L154C14-L154C20) instead of the `--root` [option](https://github.com/aurutils/aurutils/blob/fae585e740703786423803ef33336d8924192fd4/lib/aur-build#L127). `aur-sync`'s manual suggests `--root` is a valid option.

The following command:
```
aur sync --no-ver --no-confirm --no-view --database REDACTED_REPO --root /home/build/database yay
# leads to 
++(/usr/lib/aurutils/aur-sync:279): aur build --syncdeps --noconfirm -d REDACTED_REPO -r /home/build/database --status
# which leads to
++(/usr/lib/aurutils/aur-build:261): aur repo --repo REDACTED_REPO --status
```
Fails with:
```
+(/usr/lib/aurutils/aur-repo:197): printf '%s: %q: object is remote (use -S to query)\n' repo https://SOME_S3_BUCKET-2.amazonaws.com/REDACTED_REPO-arch/repo/x86_x64
```
Modifying the code locally as suggested runs successfully.

Thanks for the great project! I've used it for a couple of years smoothly without having to touch the script I wrote a while ago. So I may be mislead in how I'm using it. Regardless, the config option issue seems legit.

**Comment by AladW** [comment]
Fixed in 19.1

**Comment by paul-ri** [comment]
Wohoo, thanks!
